app.py

```
@@ -34,25 +41,200 @@
+                # Success: Log the user in
+                session.clear() # Clear any old session data
+                session['user_id'] = teacher['id']
+                session['username'] = teacher['username'] # Store username in session too for display
+                print(f"Login successful for {username}.")
+                flash(f"Logged in successfully as {teacher['username']}.", "success")
+
+                # Redirect to the next URL if stored, otherwise to index
+                # next_url = session.pop('next_url', None) # Example of using stored next_url
+                # return redirect(next_url or url_for('index'))
+                return redirect(url_for('index'))
+            else:
```

Ruff (0.8.2)

Unnecessary `else` after `return` statement

Remove unnecessary `else`

(RET505)